Modules/_ssl.c

```diff
@@ -1050,6 +1050,107 @@ _ssl__SSLSocket_do_handshake_impl(PySSLSocket *self)
     return NULL;
 }
 
+#if defined(TLS1_3_VERSION) && !defined(OPENSSL_NO_TLS1_3)
```

Comment:
I think you need to provide alternative implementations of these functions raising `NotImplementedError` for the `#else` branch (otherwise this might not compile at all).

Reply:
Make sense, will do.

Reply:
Actually the `#if` should be inside the function, otherwise argument clinic will have hard time figuring out what you are doing.

Reply:
Yeah it'll be safer that way, will do (currently it seems clinic is clever enough to copy the `ifdef` to `.h` file, and make it an empty `METHODDEF` macro if false, so that the same `ifdef` won't be necessary when using the `METHODDEF`).

Reply:
> currently it seems clinic is clever enough to copy the ifdef to .h file, and make it an empty METHODDEF macro if false, so that the same ifdef won't be necessary when using the METHODDEF

That's cool, I didn't know it's that smart :) Than feel free to keep your code as is.